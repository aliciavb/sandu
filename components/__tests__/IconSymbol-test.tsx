import * as React from 'react';
import { render } from '@testing-library/react-native';
import { IconSymbol } from '../ui/IconSymbol';

describe('IconSymbol', () => {
  it('renders home icon correctly', () => {
    const { toJSON } = render(
      <IconSymbol 
        name="house.fill"
        size={24}
        color="black"
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders with custom size', () => {
    const { toJSON } = render(
      <IconSymbol 
        name="paperplane.fill"
        size={32}
        color="blue"
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders with custom style', () => {
    const { toJSON } = render(
      <IconSymbol 
        name="chevron.right"
        color="red"
        style={{ marginLeft: 10 }}
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });
}); 