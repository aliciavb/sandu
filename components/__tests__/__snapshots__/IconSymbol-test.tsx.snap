// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconSymbol renders home icon correctly 1`] = `
<ViewManagerAdapter_SymbolModule
  animated={false}
  colors={[]}
  name="house.fill"
  resizeMode="scaleAspectFit"
  style={
    [
      {
        "height": 24,
        "width": 24,
      },
      [
        {
          "height": 24,
          "width": 24,
        },
        undefined,
      ],
    ]
  }
  tintColor="black"
  type="monochrome"
  weight="regular"
/>
`;

exports[`IconSymbol renders with custom size 1`] = `
<ViewManagerAdapter_SymbolModule
  animated={false}
  colors={[]}
  name="paperplane.fill"
  resizeMode="scaleAspectFit"
  style={
    [
      {
        "height": 24,
        "width": 24,
      },
      [
        {
          "height": 32,
          "width": 32,
        },
        undefined,
      ],
    ]
  }
  tintColor="blue"
  type="monochrome"
  weight="regular"
/>
`;

exports[`IconSymbol renders with custom style 1`] = `
<ViewManagerAdapter_SymbolModule
  animated={false}
  colors={[]}
  name="chevron.right"
  resizeMode="scaleAspectFit"
  style={
    [
      {
        "height": 24,
        "width": 24,
      },
      [
        {
          "height": 24,
          "width": 24,
        },
        {
          "marginLeft": 10,
        },
      ],
    ]
  }
  tintColor="red"
  type="monochrome"
  weight="regular"
/>
`;
